tests/acceptance/features/bootstrap/NotificationContext.php

```diff
@@ -211,6 +340,22 @@ public function userShouldGetANotificationWithMessage(string $user, string $subj
 		);
 	}
 
+	/**
+	 * @Then user :user should not have a notification related to resource :resource with subject :subject
+	 *
+	 * @param string $user
+	 * @param string $resource
+	 * @param string $subject
+	 *
+	 * @return void
+	 */
+	public function userShouldNotHaveANotificationRelatedToResourceWithSubject(string $user, string $resource, string $subject):void {
+		$this->userListAllNotifications($user);
+		$this->featureContext->theHTTPStatusCodeShouldBe(200);
+		$response = $this->filterResponseByNotificationSubjectAndResource($subject, $resource);
```

Comment:
And here the same function checks the empty array. We can make this function return boolean and function name can be renamed.

Reply:
ok think of a case where multiple notifications come with the same source and resource then will return true or false seen nice. Does this clear your confusion about not using a boolean? And the point is returning true and false making us hard to debug intermittent failure occurring in CI.